src/components/BookDetail/BookDetail.jsx

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useState } from "react"
+import { useParams } from "react-router-dom"
+
+import "./BookDetail.css"
+
+function BookDetail() {
+
+    const {id} = useParams();
+
+    const [bookImage, setBookImage] = useState('')
+    const [bookTitle, setBookTitle] = useState('')
+    const [bookAuthor, setBookAuthor] = useState('')
+    const [bookYear, setBookYear] = useState('')
+    const [bookPageCount, setBookPageCount] = useState('')
+    const [bookGenre, setBookGenre] = useState('')
+    const [bookBlurb, setBookBlurb] = useState('')
+
+    const[error, setError] = useState(false)
+
+    //Fetch the individual book data
+    useEffect(function() {
+
+    fetch(`https://book-swap-api.dev.io-academy.uk/api/books/${id}`, {
+        mode: 'cors',
+        headers: {
+            "Content-Type": "application/json",
+            "Accept": "application/json"
+        },
+        method: "GET",
+    }).then(res => res.json())
+        .then(bookData => {
+
+            // If bookData has a key of data, we know the request worked
+            if('data' in bookData)  {
+                setBookImage(bookData.data.image)
+                setBookTitle(bookData.data.title)
+                setBookAuthor(bookData.data.author)
+                setBookYear(bookData.data.year)
+                setBookPageCount(bookData.data.page_count)
+                setBookGenre(bookData.data.genre.name)
+                setBookBlurb(bookData.data.blurb)
+            } else {
+                // Display an error message
+                setError(true)
+            }
+
+
+        })
```

Comment:
The indentation is not quite right here, the fetch is inside the useEffect.

Reply:
Fixed!